File: server/module/module-erp/erp-server/src/convert/erp_receipt_detail.rs
use sea_orm::{Set, NotSet};
use crate::model::erp_receipt_detail::{self, Model as ErpReceiptDetail, ActiveModel as ErpReceiptDetailActiveModel};
use erp_model::request::erp_receipt_detail::{CreateErpReceiptDetailRequest, UpdateErpReceiptDetailRequest};
use erp_model::response::erp_receipt_detail::{ErpReceiptDetailBaseResponse, ErpReceiptDetailResponse};

pub fn create_request_to_model(request: &CreateErpReceiptDetailRequest) -> ErpReceiptDetailActiveModel {
    ErpReceiptDetailActiveModel {
        sales_order_id: request.sales_order_id.as_ref().map_or(NotSet, |sales_order_id| Set(Some(sales_order_id.clone()))),
        sales_return_id: request.sales_return_id.as_ref().map_or(NotSet, |sales_return_id| Set(Some(sales_return_id.clone()))),
        amount: Set(request.amount.clone()),
        remarks: request.remarks.as_ref().map_or(NotSet, |remarks| Set(Some(remarks.clone()))),
        ..Default::default()
    }
}

pub fn update_add_request_to_model(request: &UpdateErpReceiptDetailRequest) -> ErpReceiptDetailActiveModel {
    ErpReceiptDetailActiveModel {
        sales_order_id: request.sales_order_id.as_ref().map_or(NotSet, |sales_order_id| Set(Some(sales_order_id.clone()))),
        sales_return_id: request.sales_return_id.as_ref().map_or(NotSet, |sales_return_id| Set(Some(sales_return_id.clone()))),
        amount: Set(request.amount.clone()),
        remarks: request.remarks.as_ref().map_or(NotSet, |remarks| Set(Some(remarks.clone()))),
        ..Default::default()
    }
}

pub fn update_request_to_model(request: &UpdateErpReceiptDetailRequest, existing: ErpReceiptDetail) -> ErpReceiptDetailActiveModel {
    let mut active_model: ErpReceiptDetailActiveModel = existing.into();
    if let Some(sales_order_id) = &request.sales_order_id { 
        active_model.sales_order_id = Set(Some(sales_order_id.clone()));
    }
    if let Some(sales_return_id) = &request.sales_return_id { 
        active_model.sales_return_id = Set(Some(sales_return_id.clone()));
    }
    if let amount = &request.amount { 
        active_model.amount = Set(amount.clone());
    }
    if let Some(remarks) = &request.remarks { 
        active_model.remarks = Set(Some(remarks.clone()));
    }
    active_model
}

pub fn model_to_response(model: ErpReceiptDetail) -> ErpReceiptDetailResponse {
    ErpReceiptDetailResponse { 
        id: model.id,
        receipt_id: model.receipt_id,
        sales_order_id: model.sales_order_id,
        sales_return_id: model.sales_return_id,
        amount: model.amount,
        remarks: model.remarks,
        department_code: model.department_code,
        department_id: model.department_id,
        creator: model.creator,
        create_time: model.create_time,
        updater: model.updater,
        update_time: model.update_time,
    }
}

pub fn model_to_base_response(model: ErpReceiptDetail) -> ErpReceiptDetailBaseResponse {
    ErpReceiptDetailBaseResponse { 
        id: model.id,
        sales_order_id: model.sales_order_id,
        sales_return_id: model.sales_return_id,
        amount: model.amount,
        remarks: model.remarks,
    }
}

File: admin/src/api/erp_receipt_detail.ts
import { PaginatedRequest, PaginatedResponse } from "@/base/page";
import { api } from "@/utils/request";

const apis = {
  create: "/erp/erp_receipt_detail/create", // 新增
  update: "/erp/erp_receipt_detail/update", // 修改
  delete: "/erp/erp_receipt_detail/delete", // 删除
  get: "/erp/erp_receipt_detail/get", // 单条查询
  list: "/erp/erp_receipt_detail/list", // 列表查询
  page: "/erp/erp_receipt_detail/page", // 分页查询
};

export interface ErpReceiptDetailRequest {
  id?: number; // 收款详情ID
  sales_order_id: number; // 销售订单ID
  sales_return_id: number; // 销售退货ID
  amount: number; // 金额
  remarks: string; // 备注
}

export interface ErpReceiptDetailResponse {
  id: number; // 收款详情ID
  receipt_id: number; // 收款ID
  sales_order_id: number; // 销售订单ID
  sales_return_id: number; // 销售退货ID
  amount: number; // 金额
  remarks: string; // 备注
  department_code: string; // 部门编码
  department_id: number; // 部门ID
  creator: number; // 创建者ID
  create_time: string; // 创建时间
  updater: number; // 更新者ID
  update_time: string; // 更新时间
}

export interface ErpReceiptDetailBaseResponse {
  id: number; // 收款详情ID
  sales_order_id: number; // 销售订单ID
  sales_return_id: number; // 销售退货ID
  amount: number; // 金额
  remarks: string; // 备注
}

export interface ErpReceiptDetailQueryCondition extends PaginatedRequest {}

export const createErpReceiptDetail = (
  erp_receipt_detail: ErpReceiptDetailRequest
): Promise<number> => {
  return api.post<number>(apis.create, erp_receipt_detail);
};

export const updateErpReceiptDetail = (
  erp_receipt_detail: ErpReceiptDetailRequest
): Promise<void> => {
  return api.post<void>(apis.update, erp_receipt_detail);
};

export const deleteErpReceiptDetail = (id: number): Promise<void> => {
  return api.post<void>(`${apis.delete}/${id}`);
};

export const getErpReceiptDetail = (
  id: number
): Promise<ErpReceiptDetailResponse> => {
  return api.get<ErpReceiptDetailResponse>(`${apis.get}/${id}`);
};

export const listErpReceiptDetail = (): Promise<
  Array<ErpReceiptDetailResponse>
> => {
  return api.get<Array<ErpReceiptDetailResponse>>(apis.list);
};

export const pageErpReceiptDetail = (
  condition: ErpReceiptDetailQueryCondition
): Promise<PaginatedResponse<ErpReceiptDetailResponse>> => {
  return api.get<PaginatedResponse<ErpReceiptDetailResponse>>(
    apis.page,
    condition
  );
};
